Update docs for earthly config (#104)
earthly/earthly: docs/earth-config/earth-config.md

# Earthly config file

Global configuration values for earth can be stored on disk in the configuration file.

By default, earth reads the configuration file `~/.earthly/config.yaml`; however, it can also be
overridden with the `--config` command flag option.

> Currently only https authentication configuration is supported by the config file.

Reviewer: Perhaps this should be clarified that only https auth is *configured* here, however both SSH and user-pass auth are supported. We should also point to the auth guide for more info.

## Format

The earthly config file is a [yaml](https://yaml.org/) formatted file that looks like:

```yaml
git:
    <site>:

>         auth: https

Reviewer: ```suggestion
        auth: https|ssh
```

        user: <username>
        password: <password>
    <site2>:
        ...

> ```

Reviewer: Might be worth also adding a `github.com` example just below this format.

Author: excellent point

### HTTPS Authenticating

To configure https authentication, the auth mode must be set to https
in addition to the user and password values.

### SSH Authenticating

By default earth uses ssh authentication. It can be explicitly set by setting `auth: ssh`,
or implicitly set by omitting the site-specific configuration.

> See the [Authenticat guide](guides/auth.md) for more details.

Reviewer: It feels like this is more like a guide rather than reference. I would keep the auth guide page to serve as the guide for these settings.

For this particular page, I would just keep it purely a reference page, with just format and reference. See the [docker-compose reference](https://docs.docker.com/compose/compose-file/) as an example.

Perhaps this could be the structure, roughly:

```markdown
## Format
...
## Global configuration reference (future)
...
### some-config-item
...
### another-config-item
...
## Git configuration reference
...
### auth
...
### user
...
### password
...
```

Author: thanks for the example, I've fixed this up.

# Earthly configuration file

Global configuration values for earth can be stored on disk in the configuration file.

By default, earth reads the configuration file `~/.earthly/config.yaml`; however, it can also be
overridden with the `--config` command flag option.

Currently only https authentication configuration is configured in the config file; ssh authentication

> is additionally supported (and the default when the no overrides have been configured)

Reviewer: ```suggestion
is additionally [supported](../guides/auth.md) (and the default when the no overrides have been configured).
```

## Format

The earthly config file is a [yaml](https://yaml.org/) formatted file that looks like:

```yaml
git:
    <site>:
        auth: https|ssh
        user: <username>
        password: <password>
    <site2>:
        ...
```

Example:

```yaml
git:
    github.com:
        auth: https
        user: alice
        password: itsasecret
```

## Git configuration reference

### site

> The git repo hostname; e.g. `github.com`, or `gitlab.com`

Reviewer: ```suggestion
The git repository hostname. For example `github.com`, or `gitlab.com`
```

### auth

Either `https` or `ssh` (default). If https is specified, user and password fields are used
to authenticate over https when pulling from git for the corresponding site.

> See the [Authentication guide](guides/auth.md) for a guide on setting up authentication.

Reviewer: ```suggestion
See the [Authentication guide](../guides/auth.md) for a guide on setting up authentication.
```

### user

> The https username to use when auth is set to https; this is ignored when auth is ssh.

Reviewer: ```suggestion
The https username to use when auth is set to `https`. This setting is ignored when auth is `ssh`.
```

### password

> The https password to use when auth is set to https; this is ignored when auth is ssh.

Reviewer: The https password to use when auth is set to `https`. This setting is ignored when auth is `ssh`.
